src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs

```diff
@@ -66,6 +68,8 @@ public AttributeRouteModel(AttributeRouteModel other)
 
         public bool IsAbsoluteTemplate => Template != null && IsOverridePattern(Template);
 
+        public IParameterTransformer RouteTokenTransformer { get; set; }
```

Comment:
Just wanted to point out that this API shape implies a single parameter transformer for all of the 'mvc' parameters. This is probably the most common usage, but the system is intended to be much more flexible.

Reply:
This design makes sense though based on the design of the `ReplaceTokens` methods.

Reply:
This value is per attribute route model. There is nothing stopping someone assigning different transformers on different attribute route model. The unit tests already do this (only applies a transformer to a single controller for testing purposes)

Reply:
Right but it's not per-parameter.

Reply:
oic, yes correct